exchanges/subscription/list.go

```diff
@@ -46,6 +46,22 @@ func (l List) GroupPairs() (n List) {
 	return s.List()
 }
 
+// GroupByPairs groups subscriptions which have the same Pairs
+func (l List) GroupByPairs() []List {
+	n := []List{}
+outer:
+	for _, a := range l {
+		for i, b := range n {
+			if a.Pairs.Equal(b[0].Pairs) {
```

Comment:
b "could be zero", any chance you could you could check length and panic with some extra info so *if* it does happen there's slightly more context?

Reply:
Maybe ... by "could be zero" you meant that if someone came in here and broke this code?

Reply:
Fixed b91e6115

panic is implicit if anyone anyone breaks this. Comment to guard against tripwire.